### ISSUE TITLE
[option] Switching case doesn't apply to the current character

### ISSUE THREAD

**Issue by louisabraham** [sspanak/tt9#181]
Hello, I'm a bit troubled by some default behavior of tt9: switching case changes the case of the *last* entered character. I think some people (including me) expect it to change the case of the *next* character. Would it be possible to add an option that activates that behavior?

**Comment by sspanak** [comment]
Hey! 

From what I understand, you would like to have the equivalent of CAPS LOCK on a computer keyboard. This is certainly possible, but changing the text case would have to happen via another key. I can't simply throw away the current functionality, because the #-key does not simply change the text case, but it changes the input mode (123, ABC, abc, Predictive). And I am sure, you would want that, too.

In short, it is possible to add a new CAPS LOCK hotkey, but changing the text case is possible even now, so I consider this issue of low priority.

**Comment by justbruce** [comment]
I seem to recall that Clam-TT9 was kinda clever about this at one point - pressing mode-change while a suggestion was active would (as TT9 currently does) cycle thru the capitalization options, but would THEN drop back to standard lowercase once the suggestion was accepted.  

This MIGHT prove a worthwhile mechanism to move since then it could just be a logic inversion and a toggle in the options. i.e. 'Capitalize previous/Capitalize Next on mode-key'

**Comment by louisabraham** [comment]
I don't think I was clear about what I meant.

Currently, if I type `2 #` it will type `a` then change its case to `A`. I think there should be an option to change that behavior to selecting the letter `a` then changing the mode to capital letters.

More generally I think that other key presses than the last key press should have the same effect as `Ok` followed by that key. Eg left and right arrows should accept the current letter then go back or forward. I'm only talking about modes `AB` and `ab` here, not the actual T9.

**Comment by sspanak** [comment]
> I seem to recall that Clam-TT9 was kinda clever about this at one point - pressing mode-change while a suggestion was active would (as TT9 currently does) cycle thru the capitalization options, but would THEN drop back to standard lowercase once the suggestion was accepted.

But this still works. In Predictive mode, the text case changes automatically, even taking into account personal name fields (where every word is supposed to start with a capital) or email/password fields, where text case must not be enforced.

> Currently, if I type 2 # it will type a then change its case to A. I think there should be an option to change that behavior to selecting the letter a then changing the mode to capital letters.

The option exists even now. Just do as you described, press `#` to change the mode and select "AB" or "ab", depending on your need.

> Eg left and right arrows should accept the current letter then go back or forward. I'm only talking about modes AB and ab here

Agreed.

**Comment by louisabraham** [comment]
> > Currently, if I type 2 # it will type a then change its case to A. I think there should be an option to change that behavior to selecting the letter a then changing the mode to capital letters.
> 
> The option exists even now. Just do as you described, press `#` to change the mode and select "AB" or "ab", depending on your need.

I think we are talking about two different things. Currently, to type `Ad` I need to press: `2, OK, #, 3`. I think this does not feel logical and would like to press `2, #, 3` and not have to type on `OK`.

**Comment by sspanak** [comment]
> Currently, to type Ad I need to press: 2, OK, #, 3

Actually, there is no need to press OK after every key. Let's say you want to type "Adam", then you would press: `2, 3#, 2 , 6, OK` to get the word. Or you could even press `0` at the end to type a space and start a new word. See, it really takes the minimum amount of key presses, it's just that you press `#` after the letter, not before it.

Now, I could technically add a hotkey for changing the text case only, but it feels redundant. It will not speed typing and it just copies existing functionality...
